nodebuilder/fraud/service.go

```diff
@@ -2,23 +2,89 @@ package fraud
 
 import (
 	"context"
+	"encoding/json"
 
 	"github.com/celestiaorg/celestia-node/fraud"
 )
 
-// Module encompasses the behavior necessary to subscribe and broadcast fraud proofs within the
-// network. Any method signature changed here needs to also be changed in the API struct.
-type Module interface {
+var _ Module = (*Service)(nil)
+
+// Service is an implementation of Module that uses fraud.Service as a backend. It is used to
+// provide fraud proofs as a non-interface type to the API, and wrap fraud.Subscriber with a
+// channel of Proofs.
+type Service struct {
 	fraud.Service
 }
 
-// API is a wrapper around Module for the RPC.
-// TODO(@distractedm1nd): These structs need to be autogenerated.
-//
-// Module
-//
-//go:generate go run github.com/golang/mock/mockgen -destination=mocks/api.go -package=mocks .
-type API struct {
-	Subscribe func(fraud.ProofType) (fraud.Subscription, error)
-	Get       func(context.Context, fraud.ProofType) ([]fraud.Proof, error)
+func (s *Service) Subscribe(ctx context.Context, proofType fraud.ProofType) (<-chan Proof, error) {
+	subscription, err := s.Service.Subscribe(proofType)
+	if err != nil {
+		return nil, err
+	}
+	proofs := make(chan Proof)
+	go func() {
+		defer close(proofs)
+		for {
+			proof, err := subscription.Proof(ctx)
+			if err != nil {
+				if err != context.DeadlineExceeded && err != context.Canceled {
+					log.Errorw("fetching proof from subscription", "err", err)
+				}
+				return
+			}
+			select {
+			case <-ctx.Done():
+				return
+			case proofs <- Proof{Proof: proof}:
+			}
+		}
+	}()
+	return proofs, nil
+}
+
+func (s *Service) Get(ctx context.Context, proofType fraud.ProofType) ([]Proof, error) {
+	originalProofs, err := s.Service.Get(ctx, proofType)
+	if err != nil {
+		return nil, err
+	}
+	proofs := make([]Proof, len(originalProofs))
+	for i, originalProof := range originalProofs {
+		proofs[i].Proof = originalProof
+	}
+	return proofs, nil
+}
+
+// Proof embeds the fraud.Proof interface type to provide a concrete type for JSON serialization.
+type Proof struct {
+	fraud.Proof
+}
+
+type fraudProof struct {
```

Comment:
optional nit for clarity:
```suggestion
type fraudProofJSON struct {
```

Reply:
done